tests/integration/components/moderators/component-test.ts

```diff
@@ -0,0 +1,136 @@
+        this.provider.update({
+            moderators: [...moderators],
+        });
+        await render(
+            hbs`<Moderators::Manager
+                    @provider={{this.providerModel}}
+                    as |moderatorManager|
+                >
+                    <Moderators::List
+                        @manager={{moderatorManager}}
+                    />
+                </Moderators::Manager>`,
+        );
+        assert.dom('[data-test-moderator-link]').exists({ count: 10 });
+        assert.dom('[data-test-permission-group]').exists({ count: 10 });
+        await click('[data-test-next-page-button]');
+        assert.dom('[data-test-moderator-link]').exists({ count: 8 });
+        assert.dom('[data-test-permission-group]').exists({ count: 8 });
+    });
+
+    test('can update user permission and remove moderator as an admin', async function(
+        this: ModeratorsTestContext, assert,
+    ) {
+        const currentUser = server.create('user', { id: 'sprout' });
+        const currentUserModel = await this.store.findRecord('user', 'sprout');
+        this.owner.lookup('service:current-user').setProperties({
+            user: currentUserModel, currentUserId: currentUserModel.id,
+        });
+        const moderator = server.create('moderator', {
+            permissionGroup: PermissionGroup.Moderator,
+        });
+        const admin = server.create('moderator', {
+            id: currentUserModel.id,
+            user: currentUser,
+            permissionGroup: PermissionGroup.Admin,
+        });
+        this.provider.update({
+            moderators: [moderator, admin],
+        });
+        await render(
+            hbs`<Moderators::Manager
+                    @provider={{this.providerModel}}
+                    as |moderatorManager|
+                >
+                    <Moderators::List
+                        @manager={{moderatorManager}}
+                    />
+                </Moderators::Manager>`,
+        );
+        assert.dom('[data-test-moderator-link]').exists({ count: 2 });
+        assert.dom('[data-test-permission-group]').exists({ count: 2 });
+        assert.dom('[data-test-delete-moderator-button]').exists({ count: 2 });
+        assert.dom(`[data-test-moderator-row="${currentUser.id}"]>div>[data-test-permission-group]`).hasText('Admin');
+        assert.dom(`[data-test-moderator-row="${moderator.id}"]>div>[data-test-permission-group]`).hasText('Moderator');
+        await clickTrigger(`[data-test-moderator-row="${moderator.id}"]`);
+        await selectChoose(`[data-test-moderator-row="${moderator.id}"]`, 'Admin');
+        assert.dom(`[data-test-moderator-row="${currentUser.id}"]>div>[data-test-permission-group]`).hasText('Admin');
+        assert.dom(`[data-test-moderator-row="${moderator.id}"]>div>[data-test-permission-group]`).hasText('Admin');
+        await click(`[data-test-delete-moderator-button="${moderator.id}"]`);
```

Comment:
As you said earlier, now that we have a modal when deleting, this will need to be updated a little. The delete button on the moderator row should have this tag: `data-test-delete-button`, the dialog that shows up will ask: `Are you sure you want to remove ${moderator.fullName}?`, and the Delete button on the modal will have this tag: `data-test-confirm-delete`